### PR TITLE
DOCS-21596: Kafka REST examples (kafka topic consume and produce)

### PR DESCRIPTION
**Pull request by tikimims**
Release Notes
-------------
New Features
- Add Kafka REST examples to ``confluent kafka topic produce`` and ``confluent kafka topic consume``

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Add more Kafka Rest examples

References
----------
- DOCS-JIRA: https://confluentinc.atlassian.net/browse/DOCS-21596

Test & Review
-------------

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/cmd/kafka/command_topic_consume_onprem.go [hunk 1]
```diff
@@ -25,6 +25,14 @@ func (c *authenticatedTopicCommand) newConsumeCommandOnPrem() *cobra.Command {
 		Short: "Consume messages from a Kafka topic.",
 		Long:  "Consume messages from a Kafka topic. Configuration and command guide: https://docs.confluent.io/confluent-cli/current/cp-produce-consume.html.\n\nTruncated message headers will be printed if they exist.",
 		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Consume message from topic "my_topic" providing embedded Kafka REST Proxy endpoint.`,
+				Code: "confluent kafka topic consume my_topic --url http://localhost:8090/kafka",
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_consume_onprem.go:30]: While we're at it, do you mind changing all instances of `my_topic` to `my-topic`? We'd like to get rid of mixed dashes and underscores to make things easier to type!
```suggestion
				Text: `Consume message from topic "my-topic" providing embedded Kafka REST Proxy endpoint.`,
				Code: "confluent kafka topic consume my-topic --url http://localhost:8090/kafka",
```

**Review by brianstrauch — APPROVED** [review]

File: internal/cmd/kafka/command_topic_consume_onprem.go (hunk 1 above)
Comment (tikimims) [reply internal/cmd/kafka/command_topic_consume_onprem.go:30]: Okay got it. Will go in and update those instances 👍 Thanks @brianstrauch

File: internal/cmd/kafka/command_topic_consume_onprem.go [hunk 2]
```diff
@@ -26,11 +26,19 @@ func (c *authenticatedTopicCommand) newConsumeCommandOnPrem() *cobra.Command {
 		Long:  "Consume messages from a Kafka topic. Configuration and command guide: https://docs.confluent.io/confluent-cli/current/cp-produce-consume.html.\n\nTruncated message headers will be printed if they exist.",
 		Example: examples.BuildExampleString(
 			examples.Example{
-				Text: `Consume message from topic "my_topic" with SSL protocol and SSL verification enabled (providing certificate and private key).`,
-				Code: `confluent kafka topic consume my_topic --protocol SSL --bootstrap "localhost:19091" --ca-location my-cert.crt --cert-location client.pem --key-location client.key`},
+				Text: `Consume message from topic "my-topic" providing embedded Kafka REST Proxy endpoint.`,
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_consume_onprem.go:29]:
```suggestion
				Text: `Consume messages from topic "my-topic", providing embedded Kafka REST Proxy endpoint.`,
```

File: internal/cmd/kafka/command_topic_consume_onprem.go [hunk 3]
```diff
@@ -26,11 +26,19 @@ func (c *authenticatedTopicCommand) newConsumeCommandOnPrem() *cobra.Command {
...
+				Code: "confluent kafka topic consume my-topic --url http://localhost:8090/kafka",
+			},
 			examples.Example{
-				Text: `Consume message from topic "my_topic" with SASL_SSL/OAUTHBEARER protocol enabled (using MDS token).`,
-				Code: `confluent kafka topic consume my_topic --protocol SASL_SSL --sasl-mechanism OAUTHBEARER --bootstrap "localhost:19091" --ca-location my-cert.crt`},
+				Text: `Consume message from topic "my-topic" providing Kafka REST Proxy endpoint.`,
+				Code: "confluent kafka topic consume my-topic --url http://localhost:8082",
+			},
+			examples.Example{
+				Text: `Consume message from topic "my-topic" with SSL protocol and SSL verification enabled (providing certificate and private key).`,
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_consume_onprem.go:37]:
```suggestion
				Text: `Consume messages from topic "my-topic" with SSL protocol and SSL verification enabled, providing certificate and private key.`,
```

**Comment by tikimims** [comment]
@brianstrauch Can you help with this one? I've been checking back on it. From my end, it looks like its been building for awhile to meet the checks. 😅 Is this expected?

File: internal/cmd/kafka/command_topic_consume_onprem.go [hunk 4]
```diff
@@ -26,11 +26,19 @@ func (c *authenticatedTopicCommand) newConsumeCommandOnPrem() *cobra.Command {
...
 			examples.Example{
-				Text: `Consume message from topic "my_topic" with SSL protocol and SSL verification enabled (providing certificate and private key).`,
-				Code: `confluent kafka topic consume my_topic --protocol SSL --bootstrap "localhost:19091" --ca-location my-cert.crt --cert-location client.pem --key-location client.key`},
+				Text: `Consume messages from topic "my-topic", providing embedded Kafka REST Proxy endpoint.`,
+				Code: "confluent kafka topic consume my-topic --url http://localhost:8090/kafka",
+			},
 			examples.Example{
-				Text: `Consume message from topic "my_topic" with SASL_SSL/OAUTHBEARER protocol enabled (using MDS token).`,
-				Code: `confluent kafka topic consume my_topic --protocol SASL_SSL --sasl-mechanism OAUTHBEARER --bootstrap "localhost:19091" --ca-location my-cert.crt`},
+				Text: `Consume message from topic "my-topic" providing Kafka REST Proxy endpoint.`,
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_consume_onprem.go:33]:
```suggestion
				Text: `Consume messages from topic "my-topic" providing Kafka REST Proxy endpoint.`,
```

File: internal/cmd/kafka/command_topic_consume_onprem.go [hunk 5]
```diff
@@ -26,11 +26,19 @@ func (c *authenticatedTopicCommand) newConsumeCommandOnPrem() *cobra.Command {
...
+				Code: "confluent kafka topic consume my-topic --url http://localhost:8082",
+			},
+			examples.Example{
+				Text: `Consume messages from topic "my-topic" with SSL protocol and SSL verification enabled, providing certificate and private key.`,
+				Code: `confluent kafka topic consume my-topic --protocol SSL --bootstrap "localhost:19091" --ca-location my-cert.crt --cert-location client.pem --key-location client.key`},
+			examples.Example{
+				Text: `Consume message from topic "my-topic" with SASL_SSL/OAUTHBEARER protocol enabled (using MDS token).`,
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_consume_onprem.go:40]:
```suggestion
				Text: `Consume messages from topic "my-topic" with SASL_SSL/OAUTHBEARER protocol enabled (using MDS token).`,
```

File: internal/cmd/kafka/command_topic_produce_onprem.go [hunk 6]
```diff
@@ -27,12 +27,20 @@ func (c *authenticatedTopicCommand) newProduceCommandOnPrem() *cobra.Command {
 		Long:  "Produce messages to a Kafka topic. Configuration and command guide: https://docs.confluent.io/confluent-cli/current/cp-produce-consume.html.\n\nWhen using this command, you cannot modify the message header, and the message header will not be printed out.",
 		Example: examples.BuildExampleString(
 			examples.Example{
-				Text: `Produce message to topic "my_topic" with SASL_SSL/PLAIN protocol (providing username and password).`,
-				Code: `confluent kafka topic produce my_topic --protocol SASL_SSL --sasl-mechanism PLAIN --bootstrap "localhost:19091" --username user --password secret --ca-location my-cert.crt`,
+				Text: `Produce message to topic "my-topic" providing embedded Kafka REST Proxy endpoint.`,
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_produce_onprem.go:30]:
```suggestion
				Text: `Produce messages to topic "my-topic" providing embedded Kafka REST Proxy endpoint.`,
```

File: internal/cmd/kafka/command_topic_produce_onprem.go [hunk 7]
```diff
@@ -27,12 +27,20 @@ func (c *authenticatedTopicCommand) newProduceCommandOnPrem() *cobra.Command {
...
+				Code: "confluent kafka topic produce my-topic --url http://localhost:8090/kafka",
 			},
 			examples.Example{
-				Text: `Produce message to topic "my_topic" with SSL protocol, and SSL verification enabled.`,
-				Code: `confluent kafka topic produce my_topic --protocol SSL --bootstrap "localhost:18091" --ca-location my-cert.crt`,
+				Text: `Produce message to topic "my-topic" providing Kafka REST Proxy endpoint.`,
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_produce_onprem.go:34]:
```suggestion
				Text: `Produce messages to topic "my-topic" providing Kafka REST Proxy endpoint.`,
```

File: internal/cmd/kafka/command_topic_produce_onprem.go [hunk 8]
```diff
@@ -27,12 +27,20 @@ func (c *authenticatedTopicCommand) newProduceCommandOnPrem() *cobra.Command {
...
+				Code: "confluent kafka topic produce my-topic --url http://localhost:8082",
+			},
+			examples.Example{
+				Text: `Produce message to topic "my-topic" with SASL_SSL/PLAIN protocol (providing username and password).`,
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_produce_onprem.go:38]:
```suggestion
				Text: `Produce messages to topic "my-topic" with SASL_SSL/PLAIN protocol (providing username and password).`,
```

File: internal/cmd/kafka/command_topic_produce_onprem.go [hunk 9]
```diff
@@ -27,12 +27,20 @@ func (c *authenticatedTopicCommand) newProduceCommandOnPrem() *cobra.Command {
...
+				Code: `confluent kafka topic produce my-topic --protocol SASL_SSL --sasl-mechanism PLAIN --bootstrap "localhost:19091" --username user --password secret --ca-location my-cert.crt`,
+			},
+			examples.Example{
+				Text: `Produce message to topic "my-topic" with SSL protocol, and SSL verification enabled.`,
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_produce_onprem.go:42]:
```suggestion
				Text: `Produce messages to topic "my-topic" with SSL protocol, and SSL verification enabled.`,
```

**Comment by brianstrauch** [comment]
@tikimims The errors are expected. https://github.com/confluentinc/cli/pull/1716 fixes this, but I'm waiting on a review from @confluentinc/cli.

**Comment by tikimims** [comment]
> @tikimims The errors are expected. #1716 fixes this, but I'm waiting on a review from @confluentinc/cli.

Ahh I see. Good to know. Thank you! 👍

**Comment by tikimims** [comment]
@brianstrauch :) Since https://github.com/confluentinc/cli/pull/1716 is closed/fixed. Is there something else that I need to update here for the checks to pass here?

**Comment by brianstrauch** [comment]
Looks like the CI job is failing on `make lint`! @tikimims

**Comment by tikimims** [comment]
> Looks like the CI job is failing on `make lint`! @tikimims



> Looks like the CI job is failing on `make lint`! @tikimims

Okay ran _make lint_ locally and will update. (FYI still getting "WARNING: Failed to fetch feature flags.

Suggestions:
    Check connectivity to https://confluent.cloud or set "disable_feature_flags": true in ~/.confluent/config.json." when running _make lint_ but now able to see some errors.

File: internal/cmd/kafka/command_topic_consume_onprem.go (hunk 2 above)
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_consume_onprem.go:29]: Looks like this change got reverted by accident?
```suggestion
				Text: `Consume messages from topic "my-topic" providing embedded Kafka REST Proxy endpoint.`,
```

File: internal/cmd/kafka/command_topic_consume_onprem.go [hunk 10]
```diff
@@ -26,11 +26,19 @@ func (c *authenticatedTopicCommand) newConsumeCommandOnPrem() *cobra.Command {
...
+				Code: `confluent kafka topic consume my-topic --url http://localhost:8090/kafka --bootstrap "localhost:19091" --ca-location my-cert.crt`,
+			},
 			examples.Example{
-				Text: `Consume message from topic "my_topic" with SASL_SSL/OAUTHBEARER protocol enabled (using MDS token).`,
-				Code: `confluent kafka topic consume my_topic --protocol SASL_SSL --sasl-mechanism OAUTHBEARER --bootstrap "localhost:19091" --ca-location my-cert.crt`},
+				Text: `Consume messages from topic "my-topic" providing Kafka REST Proxy endpoint.`,
+				Code: `confluent kafka topic consume my-topic --url http://localhost:8082 --bootstrap "localhost:19091" --ca-location my-cert.crt`,
+			},
+			examples.Example{
+				Text: `Consume messags from topic "my-topic" with SSL protocol and SSL verification enabled (providing certificate and private key).`,
```
Comment (tikimims) [inline internal/cmd/kafka/command_topic_consume_onprem.go:37]:
```suggestion
				Text: `Consume messages from topic "my-topic" with SSL protocol and SSL verification enabled (providing certificate and private key).`,
```

File: internal/cmd/kafka/command_topic_consume_onprem.go (hunk 2 above)
Comment (tikimims) [reply internal/cmd/kafka/command_topic_consume_onprem.go]: Good catch. Thanks @brianstrauch

File: internal/cmd/kafka/command_topic_consume_onprem.go [hunk 11]
```diff
@@ -26,18 +26,18 @@ func (c *authenticatedTopicCommand) newConsumeCommandOnPrem() *cobra.Command {
...
 				Code: `confluent kafka topic consume my-topic --url http://localhost:8090/kafka --bootstrap "localhost:19091" --ca-location my-cert.crt`,
 			},
 			examples.Example{
-				Text: `Consume message from topic "my-topic" providing Kafka REST Proxy endpoint.`,
+				Text: `Consume messages from topic "my-topic" providing Kafka REST Proxy endpoint.`,
 				Code: `confluent kafka topic consume my-topic --url http://localhost:8082 --bootstrap "localhost:19091" --ca-location my-cert.crt`,
 			},
 			examples.Example{
-				Text: `Consume message from topic "my-topic" with SSL protocol and SSL verification enabled (providing certificate and private key).`,
+				Text: `Consume messags from topic "my-topic" with SSL protocol and SSL verification enabled (providing certificate and private key).`,
```
Comment (brianstrauch) [inline internal/cmd/kafka/command_topic_consume_onprem.go]:
```suggestion
				Text: `Consume messages from topic "my-topic" with SSL protocol and SSL verification enabled (providing certificate and private key).`,
```

**Comment by tikimims** [comment]
Hi @brianstrauch :) Looking here (https://docs.confluent.io/confluent-cli/current/cp-produce-consume.html#producer-and-consumer-configuration-flags) and testing locally with _make lint_ and wanted to check. Is the _--url_ flag accepted with _produce_ and _consume_? It's not listed here: https://docs.confluent.io/confluent-cli/current/cp-produce-consume.html#producer-and-consumer-configuration-flags. I'm getting "unknown flag" for _--url_ when running m_ake lint_ locally.

**Comment by brianstrauch** [comment]
Good thing we have a linter set up. These commands don't use Kafka REST... so we can close out this ticket without any changes.

Sorry I didn't catch this earlier!! 🤦‍♂️

**Comment by tikimims** [comment]
> Good thing we have a linter set up. These commands don't use Kafka REST... so we can close out this ticket without any changes.
> 
> Sorry I didn't catch this earlier!! 🤦‍♂️

Thank you! :)

**Pull request closed without merging**